packages/react-scripts/scripts/build.js

```diff
@@ -57,6 +58,34 @@ measureFileSizesBeforeBuild(paths.appBuild)
     fs.emptyDirSync(paths.appBuild);
     // Merge with the public folder
     copyPublicFolder();
+
+    // Check if every vendors are defined in the package.json
+    // @remove-on-eject-begin
+    // The devDepencencies shouldn't be available for vendors
+    // But otherwise the tests fail.
+    // @remove-on-eject-end
+    const packageJson = require(paths.appPackageJson);
+    const dependencies = Object.keys(packageJson.dependencies || {})
+      .concat(Object.keys(packageJson.devDependencies || {}))
+      .concat(Object.keys(packageJson.peerDependencies || {}));
+    const vendors = extractVendors();
+    const flatVendors = [];
+    Object.keys(vendors).forEach(vendorName => {
+      flatVendors.concat(vendors[vendorName]);
+    });
```

Comment:
@Ayc0 The above didn't work for me.
I defined an array in `src/vendors.js` so `extractVendors()` was returning:
```js
{ vendors: ['react', 'react-dom'] }
```
The loop above does nothing to `flatVendor` variable (it would if it was i.e. `flatVendor.push(...vendors[vendorName])`. I fixed the loop like below:
```js
const flatVendors = Object.keys(vendors)
  .reduce(
    (acc, vendorName) => acc.concat(vendors[vendorName]),
    []
  );
```